src/Stratis.Bitcoin.IntegrationTests/NodeSyncTests.cs

```diff
@@ -311,7 +311,6 @@ public void MiningNodeWithOneConnectionAlwaysSynced()
                 Thread.Sleep(1);
 
             // Make sure that miner did not advance yet but connector did.
-            Assert.NotEqual(miner.FullNode.Chain.Tip.HashBlock, networkNode1.FullNode.Chain.Tip.HashBlock);
```

Comment:
I agree that these look a bit flaky:
- go fast and the miner will be equal to networkNode1
- go slow and it won't be equal

At the same time, a slow or fast computer will mean slow/fast propagation but also slow/fast pass through the assertions.

I think this needs to be redesigned to either wait for nodes to be in sync or only connect nodes after one of them has moved forward.